tls/s2n_resume.c

```diff
@@ -767,6 +767,40 @@ struct s2n_ticket_key *s2n_find_ticket_key(struct s2n_config *config, const uint
     return NULL;
 }
 
+struct s2n_unique_ticket_key {
+    uint8_t info[S2N_AES256_KEY_LEN];
+    uint8_t output_key[S2N_AES256_KEY_LEN];
+};
+
+/* Ensures that a session ticket encryption key is used only once per ticket.
+ *
+ * The AES-GCM encryption scheme breaks if the same nonce is used with the same key more than once.
+ * As the number of TLS connections increases per second, it becomes more probable that the same
+ * random nonce will be generated twice and used with the same ticket key.
+ * To avoid this we generate a unique session ticket encryption key for each ticket.
+ **/
+static S2N_RESULT s2n_resume_generate_unique_ticket_key(uint8_t *initial_key, struct s2n_unique_ticket_key *key)
```

Comment:
Assuming the size of data always makes me uncomfortable. Maybe leave the "initial_key"  as a blob on s2n_unique_ticket_key, but just do s2n_blob_init instead of a copy?